packages/vite/src/hbs.ts

```diff
@@ -59,17 +68,34 @@ export function hbs(): Plugin {
         }
       }
 
-      let syntheticId = needsSyntheticComponentJS(source, resolution.id);
-      if (syntheticId && isInComponents(resolution.id, resolverLoader.resolver.packageCache)) {
-        return {
-          id: syntheticId,
-          meta: {
-            'rollup-hbs-plugin': {
-              type: 'template-only-component-js',
+      if (isInComponents(resolution.id, resolverLoader.resolver.packageCache)) {
+        let syntheticId = needsSyntheticComponentJS(source, resolution.id);
+        if (syntheticId) {
+          depTracker?.trackDeps(syntheticId, [resolution.id]);
+          return {
+            id: syntheticId,
+            meta: {
+              'rollup-hbs-plugin': {
+                type: 'template-only-component-js',
+              },
             },
-          },
-        };
+          };
+        } else {
+          let correspondingHBS = syntheticJStoHBS(resolution.id);
+          if (correspondingHBS) {
+            depTracker?.trackDeps(resolution.id, [correspondingHBS]);
+          }
+        }
       }
+
+      // we should be able to clear any earlier meta by returning
+      // resolution.meta here, but vite breaks that rollup feature.
```

Comment:
Just use own cache then?
MetaCache = new Map();
And set the value for the id in the resolveId

Reply:
We could but this is working now. I checked and the rollup docs say it's OK to mutate the meta object that comes back from getModuleInfo().meta.